packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -96,7 +97,9 @@ class ReactShallowRenderer {
     if (this._instance) {
       this._updateClassComponent(element, this._context);
     } else {
-      if (shouldConstruct(element.type)) {
+      if (isForwardRef(element)) {
+        this._rendered = element.type.render(element.props, null);
```

Comment:
This seems fine in the case where you're rendering just host elements, but maybe not helpful for the HOC case where you wrap and render a single composite component...

Reply:
I think this line should be:
```js
this._rendered = element.type.render(element.props, element.ref);
```

I'm not sure I understand your thought though?

Reply:
I was thinking of this case:

```js
function withFoo(Component) {
  const Foo = (props) => <Component ref={props.forwardedRef} />
  return React.forwardRef((props, ref) => <Foo forwardedRef={ref} />)
}
shallow(<WithFooComponent><div/></WithFooComponent>)
```

You'd get a different result than without the `forwardRef` but I guess that's true of any HoC thing, might even help hide the HoC guts a bit more with is nice.

Reply:
I see. In general, I'm not sure how valuable shallow testing with anything refs or HOC related is. 😄